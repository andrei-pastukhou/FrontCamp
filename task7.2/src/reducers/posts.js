const posts = (state = [], action) => {
    switch (action.type) {
        case 'ADD_POST':
            return [
                ...state,
                {
                    id: action.id,
                    author: action.author,
                    text: action.text,
                }
            ];
        case 'DELETE_POST':
            return state.filter(({ id }) => id !== action.id);
        default:
            return state
    }
};

export default posts