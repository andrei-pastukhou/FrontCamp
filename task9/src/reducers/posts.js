const posts = (state = [], action) => {
    switch (action.type) {
        case 'ADD_POST_PENDING':
            return state;
        case 'FETCH_POST_SUCCESS':
            return action.posts;
        default:
            return state
    }
};

export default posts
