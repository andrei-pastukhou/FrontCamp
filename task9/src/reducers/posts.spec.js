import posts from './posts'

describe('posts reducer', () => {
    it('should handle initial state', () => {
        expect(
            posts(undefined, {})
        ).toEqual([])
    });
    it('should handle ADD_POST_PENDING', () => {
        expect(
        posts([], {
            type: 'ADD_POST_PENDING',
        })
        ).toEqual([])
    });
    it('should handle FETCH_POST_SUCCESS', () => {
        expect(
        posts([], {
            type: 'FETCH_POST_SUCCESS',
            posts: ['testpost1', 'testpost2']
        })
        ).toEqual( ['testpost1', 'testpost2'])
    });
});
