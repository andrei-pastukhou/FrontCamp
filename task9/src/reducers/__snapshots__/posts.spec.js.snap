// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post reduxer with snapshots  should handle ADD_POST_PENDING 1`] = `Array []`;

exports[`post reduxer with snapshots  should handle FETCH_POST_SUCCESS 1`] = `
Array [
  "testpost1",
  "testpost2",
]
`;

exports[`post reduxer with snapshots  should handle initial state 1`] = `Array []`;
